src/components/hooks/useActivityStream.js

```diff
@@ -0,0 +1,30 @@
+import {useState, useEffect} from 'react';
+import {zip} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+/**
+ * Custom hook that returns activity data associated to the room of the given ID.
+ *
+ * @param {string} roomID  ID of the room for which to return data.
+ * @param {obj} roomsAdapter  Component data adapter from which to retrieve data.
+ * @returns {Room} Activity ID associated to the room
+ */
+export default function useActivityStream(roomID, roomsAdapter) {
+  const [activityIDs, setActivityIDs] = useState({});
+
+  useEffect(() => {
+    const activityStream = zip(
```

Comment:
I think `zip` is the wrong operator here because it merges the values of the 2 observables into one as you may have noticed, e.g. `zip([1,2,3], [4,5,6]) => [(1,4),(2,5), (3,6)]`. The behavior we want is that of [`concat`](http://reactivex.io/documentation/operators/concat.html), where the data of `getPreviousRoomActivities` get send before `getRoomActivities`.

Having said that, I'm leaning towards using [`merge`](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-merge) here because in the case of observables that will come from the SDK, `getPreviousRoomActivities` will never complete and so `concat` would never send data from `getRoomActivities`. The only problem with merge is that the data could be sent out of order, e.g. user scrolls up while someone else sends a new message. I think the solution to this would be using [`useReducer`](https://reactjs.org/docs/hooks-reference.html#usereducer) instead of `useState`. The `pipe` will be unnecessary too

Reply:
this is interesting. Let me call you.